### ISSUE TITLE
[FEA] is_monotonic* properties are erroneously cached

### ISSUE THREAD

**Issue by shwina** [rapidsai/cudf#8223]
The `is_monotonic*` properties of `Column` are cached.  But because Columns are not immutable, this can lead to bugs like the following:

```python
In [5]: s
Out[5]:
0    1
1    2
2    3
3    4
4    5
dtype: int64

In [6]: s.is_monotonic_increasing
Out[6]: True

In [7]: s[2:5] = [3, 2, 1]

In [8]: s.is_monotonic_increasing
Out[8]: True

In [9]: s
Out[9]:
0    1
1    2
2    3
3    2
4    1
dtype: int64
```

Some discussion [here](https://github.com/rapidsai/cudf/pull/8130#discussion_r627619085).

**Comment by vyasr** [comment]
I will remove caching entirely in #8130, but once that PR is merged we should consider if it is possible to invalidate the cache in all cases that should, which would make caching feasible.

**Comment by shwina** [comment]
@vyasr can this issue be closed out now that #8130 is merged?

**Comment by vyasr** [comment]
Yes I think so. We can revisit better caching in the future.
